src/include/rbd/librbd.h

```diff
@@ -299,6 +299,8 @@ CEPH_RBD_API int rbd_set_image_notification(rbd_image_t image, int fd, int type)
 
 /* exclusive lock feature */
 CEPH_RBD_API int rbd_is_exclusive_lock_owner(rbd_image_t image, int *is_owner);
+CEPH_RBD_API int rbd_exclusive_lock_acquire(rbd_image_t image);
```

Comment:
I keep coming back to how we will eventually want to have a unified managed exclusive/shared lock design. How about tweaking these to be something like the following?

```
typedef enum {
  RBD_LOCK_MODE_EXCLUSIVE = 0,
  RBD_LOCK_MODE_SHARED = 1
} rbd_lock_mode_t;

CEPH_RBD_API int rbd_lock_acquire(rbd_image_t image, rbd_lock_mode_t lock_mode);
CEPH_RBD_API int rbd_lock_release(rbd_image_t image);
```

My rationale is that I envision the current ```ExclusiveLock``` and associated namespace being renamed / refactored to support shared locking as well so it would make sense to have a unified API. Internally, you can return -EOPNOTSUPP if the lock mode is shared (for now).